include/rgb666.h

```diff
@@ -53,6 +55,16 @@ class Rgb666 {
 	{
 		return !operator==(that);
 	}
+
+	// Scoped conversion helpers
+	static constexpr Rgb666 FromRgb888(const Rgb888 rgb888)
+	{
+		const auto r6 = rgb888.red >> 2;
+		const auto g6 = rgb888.green >> 2;
+		const auto b6 = rgb888.blue >> 2;
+
+		return Rgb666(r6, g6, b6);
```

Comment:
because of the bit manipulation, the `const auto` are /probably/ `int` types, so then they're being implicity cast back into the `uint8_t` in the Rgb666 constructor.

Suggest static_cast<uint8_t>(..) the RHS.